src/libraries/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs

```diff
@@ -323,6 +323,144 @@ public enum TYMED
         TYMED_ENHMF = 64,
     }
 }
+namespace System.Runtime.InteropServices.Marshalling
+{
+    [System.AttributeUsageAttribute(System.AttributeTargets.Class, Inherited = false)]
+    [System.CLSCompliantAttribute(false)]
+    public sealed partial class ComExposedClassAttribute<T> : System.Attribute, System.Runtime.InteropServices.Marshalling.IComExposedDetails where T : System.Runtime.InteropServices.Marshalling.IComExposedClass
+    {
+        public ComExposedClassAttribute() { }
+        public unsafe System.Runtime.InteropServices.ComWrappers.ComInterfaceEntry* GetComInterfaceEntries(out int count) { throw null; }
+    }
+    public sealed partial class ComObject : System.Runtime.InteropServices.IDynamicInterfaceCastable, System.Runtime.InteropServices.Marshalling.IUnmanagedVirtualMethodTableProvider
+    {
+        internal ComObject() { }
```

Comment:
I don't think this should be in the ref assembly.

Reply:
We need to have a non-public constructor here to show that the user cannot construct a `ComObject` instance themselves (no constructor here would mean an implicit public one). I generated the APIs in here with GenAPI, so they match the tool-generated ref assembly sources we use in the rest of the repo.